File: Baseball_Game/Baseball_Game.go
package main

// Runtime: 0 ms, faster than 100.00% of Go online submissions for Baseball Game.
// Memory Usage: 2.7 MB, less than 16.98% of Go online submissions for Baseball Game.




import (
	"fmt"
	"strconv"
)

func main() {
	ops := []string{"5", "2", "C", "D", "+"}
	fmt.Println(calPoints(ops))
}




func calPoints(ops []string) int {

	ls := []int{}
	sum := 0
	for _, i := range ops {
		if i != "C" && i != "D" && i != "+" {
			if s, err := strconv.Atoi(i); err == nil {
				ls = append(ls, s)
			}
		} else if i == "C" {
			ls = ls[:len(ls)-1]
		} else if i == "D" {
			ls = append(ls, 2*ls[len(ls)-1])
		} else if i == "+" {
			ls = append(ls, ls[len(ls)-1]+ls[len(ls)-2])
		}
	}
	for _, j := range ls {
		sum += j
	}
	return sum
}

File: Baseball_Game/Baseball_Game.py
'''

You are keeping score for a baseball game with strange rules. The game consists of several rounds, where the scores of past rounds may affect future rounds' scores.

At the beginning of the game, you start with an empty record. You are given a list of strings ops, where ops[i] is the ith operation you must apply to the record and is one of the following:

An integer x - Record a new score of x.
"+" - Record a new score that is the sum of the previous two scores. It is guaranteed there will always be two previous scores.
"D" - Record a new score that is double the previous score. It is guaranteed there will always be a previous score.
"C" - Invalidate the previous score, removing it from the record. It is guaranteed there will always be a previous score.
Return the sum of all the scores on the record. The test cases are generated so that the answer fits in a 32-bit integer.



'''






ops = ["5","2","C","D","+"]
def calPoints(ops):
    
    ls = []
    k = 0
    for i in range(len(ops)):
        if ops[i] != "C" and ops[i] != "D" and ops[i] != "+":
            ls.append(int(ops[i]))
        
        elif ops[i] == "C":
            ls.remove(ls[len(ls) -1] )

        elif ops[i] == "D":
            ls.append(2*ls[len(ls) -1])

        elif ops[i] == "+":
            ls.append(ls[len(ls) -1] + ls[len(ls) -2])

    return sum(ls)


print(calPoints(ops))
